src/compiler/checker.ts

```diff
@@ -31313,6 +31313,22 @@ namespace ts {
                 getNonNullableType(checkExpression(node.expression));
         }
 
+        function checkSatisfiesExpression(node: SatisfiesExpression) {
+            checkSourceElement(node.type);
+
+            const targetType = getTypeFromTypeNode(node.type);
+            if (isErrorType(targetType)) {
+                return targetType;
+            }
+
+            const exprType = getWidenedType(checkExpression(node.expression));
+            if (!isTypeIdenticalTo(targetType, exprType)) {
+                error(node.type, Diagnostics._0_is_not_satisfied_1, typeToString(targetType), typeToString(exprType));
+            }
+
+            return targetType;
+        }
+

```

Comment:
We need a new case in `getContextualType`. This code shouldn't have an implicit any:
```ts
type MyFunc = (x: string) => void;
const f = (m => m.substr(0)) satisfies MyFunc;
```